src/commons/BalanceHandler.ts

```diff
@@ -0,0 +1,32 @@
+import { Address, BigInt } from '@graphprotocol/graph-ts';
+import { getTokenDistroBalance } from '../utils/misc';
+
+export function addAllocatedTokens(
+  to: string,
+  value: BigInt,
+  tokenAddress: Address,
+  tokenDistroAddress: string,
+): void {
+  const allocatedBalance = getTokenDistroBalance(tokenAddress);
+  allocatedBalance.allocatedTokens =
+    allocatedBalance.allocatedTokens.plus(value);
+  allocatedBalance.allocationCount = allocatedBalance.allocationCount.plus(
+    BigInt.fromI32(1),
+  );
+  allocatedBalance.tokenDistroAddress = tokenDistroAddress;
+  allocatedBalance.user = to;
```

Comment:
These lines are not needed! if `getTokenDistroBalance` works correctly, returns an entity having  these fields filled.